scipy/cluster/vq.py

```diff
@@ -578,18 +598,25 @@ def _kpp(data, k, rng):
     """
 
     dims = data.shape[1] if len(data.shape) > 1 else 1
-    init = np.ndarray((k, dims))
+
+    # k should be an integer, NOT a NumPy
+    # scalar array thing...
+    if not isinstance(k, int):
+        k = xp.astype(k, xp.int64)
```

Comment:
why not leave the above 4 lines out, and below use `init = xp.empty((int(k), dims))`?